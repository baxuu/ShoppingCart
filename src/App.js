import React from 'react';

import { useSelector } from 'react-redux';

import SubmitMessage from './components/submitMessage/SubmitMessage';
import Cart from './components/Cart';

import './App.scss';

const App = () => {
  const { showCart } = useSelector(({ cart }) => cart);

  return (
    <div className='Container'>{showCart ? <Cart /> : <SubmitMessage />}</div>
  );
};

export default App;
