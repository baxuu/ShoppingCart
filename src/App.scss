@import './helpers/variables.scss';

* {
  box-sizing: border-box;
  font-family: 'Roboto', sans-serif;
  color: $color-octonary;
  font-size: 12px;
}

.Container {
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

@media (min-width: 576px) {
  .Container {
    max-width: 540px;
  }
}
@media (max-width: 768px) {
  .Container {
    max-width: 710px;
  }
}

@media (min-width: 768px) {
  .Container {
    max-width: 720px;
  }
}

@media (min-width: 992px) {
  .Container {
    max-width: 960px;
  }
}

@media (min-width: 1200px) {
  .Container {
    max-width: 1140px;
  }
}
