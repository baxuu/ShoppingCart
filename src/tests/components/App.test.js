import React from 'react';

import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

import App from '../../App';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('connect to react redux', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      products: [
        {
          id: 1,
          name: 'bike',
          image: 'image.src',
          price: 13.9,
          quantity: 1,
        },
        {
          id: 2,
          name: 'camera',
          image: 'image.src',
          price: 3.9,
          quantity: 1,
        },
      ],

      shippingCost: 23.8,
      subTotal: 11.9,
      grandTotal: 35.7,
      showCart: true,
    });
  });
  it('snapshot test', () => {
    const appWrapper = shallow(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(appWrapper).toMatchSnapshot();
  });
});
