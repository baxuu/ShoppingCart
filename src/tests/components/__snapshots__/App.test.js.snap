// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connect to react redux snapshot test 1`] = `ShallowWrapper {}`;
